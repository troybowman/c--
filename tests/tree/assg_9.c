/* errors in array indices */

int x[5];
char y[4];

int x_at(int idx) { return x[idx]; }

char y_at(int idx) { return y[idx]; }

char char_at(char string[], int idx, char x)
{
  return string[idx];
}

extern void sum(int x, int y, int z);

void main(void)
{
  x[0] = x[x_at(0)];
  x[x_at(0)] = x[0];

  y[x[x_at(y_at(char_at("test", 0)))]] = 'x';

  x[
    sum(
      x_at(1, 1),
      y_at(0, 2),
      char_at("I...", 7))
   ]   =  y[char_at("AM THE ONE WHO KNOCKS", 1)];

  x[
    sum(
      x_at(1),
      y_at(0),
      char_at("I...", 7, x[4]))
   ]   =  y[char_at("AM THE ONE WHO KNOCKS", 1)];
}
