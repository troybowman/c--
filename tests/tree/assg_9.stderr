error: expected 3 arguments for function char_at, 2 were provided. line 22
error: expected 1 arguments for function x_at, 2 were provided. line 26
error: expected 1 arguments for function y_at, 2 were provided. line 27
error: expected 3 arguments for function char_at, 2 were provided. line 28
error: line 28 - function sum called as part of an expression but does not return a value
error: expected 3 arguments for function char_at, 2 were provided. line 29
error: line 35 - function sum called as part of an expression but does not return a value
error: expected 3 arguments for function char_at, 2 were provided. line 36
c-- returned a nonzero exit code
